### ISSUE TITLE
support generic default parameters

### ISSUE THREAD

**Issue by mernxl** [gajus/eslint-plugin-flowtype#352]
Hello, I am having this issue when trying to pass defaults to generic parameters. 
```js
export class TypeComposer<TSource = any, TContext = any> {   // somehow, this works
...
203  static createTemp<TSrc = any, TCtx = any>(    // this eslint errors with message below
204      opts: TypeAsString | ComposeObjectTypeConfig<TSrc, TCtx> | GraphQLObjectType
205    ): TypeComposer<TSrc, TCtx>
206  if (!this.schemaComposer) {
...
}
```

```
ESLint: Parsing error: Unexpected token 201 | } 202 | > 203 | static createTemp<TSrc = any, TCtx = any>( | ^ 204 | opts: TypeAsString | ComposeObjectTypeConfig<TSrc, TCtx> | GraphQLObjectType 205 | ): TypeComposer<TSrc, TCtx> { 206 | if (!this.schemaComposer) {
```

**Comment by pnevyk** [comment]
Hi @mernxl, it looks like this construct doesn't work even with original Flow parser ([playground](https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoVBjGBDAzrsAMTjjAG9UwxcAXbGgSwzACNsAnAHgBUwBeMNgB2ATwB8ACgCOAVzgAuMNwCUi3hSpV2AUxoz2QsLLgBuSmAC+qK0A)) so I suppose it's not a valid syntax, though I don't know the reason. Maybe default values for generics can be used only for things like type aliases and classes? Nonetheless, this problem should be primarily discussed in Flow project.

**Comment by mernxl** [comment]
Thank you @pnevyk for the clarification. The Playground really helped. I will reference this when creating an issue at the flow project.

**Comment by pnevyk** [comment]
Note that this is problem of parsing. So if this problem is resolved in Flow, it should be then fixed in `babel-eslint`. It is not error raised from `eslint-plugin-flowtype`.

**Comment by mernxl** [comment]
Ok, thank you for the assistance.

**Comment by pnevyk** [comment]
You are welcome. Thanks for the interest.
